src/Uno.UI/Microsoft/UI/Xaml/Controls/Repeater/ItemsSourceView.cs

```diff
@@ -0,0 +1,312 @@
+				m_vector = vector;
+				ListenToCollectionChanges();
+			}
+			else
+			{
+				// The bindable interop interface are abi compatible with the corresponding
+				// WinRT interfaces.
+				var bindableVector = source as IList;
+				if (bindableVector != null)
+				{
+					m_vector.set(reinterpret_cast <const IVector<object>&> (bindableVector));
+					ListenToCollectionChanges();
+				}
+				else
+				{
+					var iterable = source.try_as<IIterable<object>>();
+					if (iterable)
+					{
+						m_vector.set(WrapIterable(iterable));
+					}
+					else
+					{
+						var bindableIterable = source.try_as<IBindableIterable>();
+						if (bindableIterable)
+						{
+							m_vector.set(WrapIterable(reinterpret_cast <const IIterable<object> &> (bindableIterable)));
+						}
+						else
+						{
+							throw hresult_invalid_argument(L"Argument 'source' is not a supported vector.");
+						}
+					}
+				}
+			}
+
+			m_uniqueIdMaping = source as IKeyIndexMapping;
+		}
+
+		public event NotifyCollectionChangedEventHandler CollectionChanged;
+
+		//#pragma region IDataSource
+
+		public int Count
+		{
+			get
+			{
+				if (m_cachedSize == -1)
+				{
+					// Call the override the very first time. After this,
+					// we can just update the size when there is a data source change.
+					m_cachedSize = GetSizeCore();
+				}
+
+				return m_cachedSize;
+			}
+		}
+
+		public object GetAt(int index)
+		{
+			return GetAtCore(index);
+		}
+
+		public bool HasKeyIndexMapping
+		{
+			get
+			{
+				return HasKeyIndexMappingCore();
+			}
+		}
+
+		public string KeyFromIndex(int index)
+		{
+			return KeyFromIndexCore(index);
+		}
+
+		public int IndexFromKey(string id)
+		{
+			return IndexFromKeyCore(id);
+		}
+
+		internal void OnItemsSourceChanged(NotifyCollectionChangedEventArgs args)
+		{
+			m_cachedSize = GetSizeCore();
+			CollectionChanged?.Invoke(this, args);
+		}
+
+
+		~InspectingDataSource()
+		{
+			UnListenToCollectionChanges();
+		}
+
+		private int GetSizeCore()
+		{
+			return m_vector.Count;
+		}
+
+		private object GetAtCore(int index)
+		{
+			return m_vector[index];
+		}
+
+		bool HasKeyIndexMappingCore()
+		{
+			return m_uniqueIdMaping != null;
+		}
+
+		string KeyFromIndexCore(int index)
+		{
+			if (m_uniqueIdMaping != null)
+			{
+				return m_uniqueIdMaping.KeyFromIndex(index);
+			}
+			else
+			{
+				throw new InvalidOperationException("ID mapping not set.");
+			}
+		}
+
+		int IndexFromKeyCore(string id)
+		{
+			if (m_uniqueIdMaping != null)
+			{
+				return m_uniqueIdMaping.IndexFromKey(id);
+			}
+			else
+			{
+				throw new InvalidOperationException("ID mapping not set.");
+			}
+		}
+
+		//#pragma endregion
+
+		private int IndexOf(object value)
+		{
+			int index = -1;
+			if (m_vector != null && value != null)
+			{
+				var v = -1;
+				v = m_vector.IndexOf(value);
+				if (v > -1)
+				{
+					index = v;
+				}
+			}
+			return index;
+		}
+
+		//IVector<object> WrapIterable(const IIterable<object>& iterable)
+		//{
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5156819)